krm-functions/lib/kubeobject/kubeobject.go

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2023 The Nephio Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package parser
+
+import (
+	"github.com/GoogleContainerTools/kpt-functions-sdk/go/fn"
+	"sigs.k8s.io/yaml"
+)
+
+type KubeObject[T1 any] struct {
+	fn.KubeObject
+}
+
+func (r *KubeObject[T1]) GetGoStruct() (T1, error) {
+	var x T1
+	if err := yaml.Unmarshal([]byte(r.String()), &x); err != nil {
+		return x, err
+	}
+	return x, nil
+}
+
+// NewFromKubeObject returns a KubeObject struct
+// It expects a fn.KubeObject as input representing the serialized yaml file
+func NewFromKubeObject[T1 any](o fn.KubeObject) *KubeObject[T1] {
```

Comment:
Probably we can check for nil input and return err. Makes it consistent with the rest .

Reply:
done